### PR TITLE
TPC: Generic Check of Pads

### PR DESCRIPTION
**Pull request by HorstMa**
This PR is a generic check of the 72 TPC pads for an expected mean value, a comparison of the mean of one pad to the global mean and a check for empty bins inside the pads

### CODE REVIEW

File: Modules/TPC/src/CheckOfPads.cxx [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+  }
+}
+
+//______________________________________________________________________________
+Quality CheckOfPads::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+  Quality result_EV = Quality::Null;
+  Quality result_Mean = Quality::Null;
+  Quality result_Global = Quality::Null;
+  Quality result_Empty = Quality::Null;
```
Comment (wiechula) [inline Modules/TPC/src/CheckOfPads.cxx:128]: Naming convention
```suggestion
  Quality resultEV = Quality::Null;
  Quality resultMean = Quality::Null;
  Quality resultGlobal = Quality::Null;
  Quality resultEmpty = Quality::Null;
```

File: Modules/TPC/src/CheckOfPads.cxx [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+          return Quality::Null;
+        }
+
+        float PadSum = 0.;
+        int PadsCount = 0;
+        float PadStdev = 0.;
+        float PadMean = 0.;
+
+        // Run twice to get the mean and the standard deviation
+        for (int RunNo = 1; RunNo <= 2; RunNo++) {
```
Comment (wiechula) [inline Modules/TPC/src/CheckOfPads.cxx:189]: Naming
```suggestion
        for (int runNo = 1; runNo <= 2; runNo++) {
```

File: Modules/TPC/src/CheckOfPads.cxx [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+          // Run1: calculate single Pad total->Mean
+          // Run2: calculate standardDeviation from mean
+          for (int xBin = 1; xBin <= MaximumXBin; xBin++) {
+            for (int yBin = 1; yBin <= MaximumYBin; yBin++) {
+              float Binvalue = h->GetBinContent(xBin, yBin);
```
Comment (wiechula) [inline Modules/TPC/src/CheckOfPads.cxx:194]:
```suggestion
              const float binvalue = h->GetBinContent(xBin, yBin);
```

File: Modules/TPC/src/CheckOfPads.cxx [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+          }
+          mSectorsQuality_Empty.push_back(result_Empty);
+          mEmptyPadPercent.push_back(1. - (float)PadsCount / totalPads);
+        }
+
+        PadStdev = sqrt(PadStdev / (PadsCount - 1));
+        mPadMeans.push_back(PadMean);
+        mPadStdev.push_back(PadStdev);
+      }
+      float SumOfWeights = 0.;
```
Comment (wiechula) [inline Modules/TPC/src/CheckOfPads.cxx:227]:
```suggestion
      float sumOfWeights = 0.;
```

File: Modules/TPC/src/CheckOfPads.cxx [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+      }
+      mTotalStdev = sqrt(1 / SumOfWeights); // standard deviation of the weighted average.
+      mTotalMean /= SumOfWeights;           // Weighted average (by standard deviation) of the total mean
+      // calculate the Qualities:
+
+      for (size_t it = 0; it < mPadMeans.size(); it++) { // loop over all pads
+
+        if (mExpectedValueCheck) {
+
+          if (fabs(mPadMeans[it] - mExpectedValue) < mPadStdev[it] * mExpectedValueMediumSigmas) {
```
Comment (wiechula) [inline Modules/TPC/src/CheckOfPads.cxx:241]: Also in all other places `fabs` -> `std::abs`
```suggestion
          if (std::abs(mPadMeans[it] - mExpectedValue) < mPadStdev[it] * mExpectedValueMediumSigmas) {
```

**Comment by Barthelemy** [comment]
is it a duplicate of https://github.com/AliceO2Group/QualityControl/pull/1404 ?

**Comment by Barthelemy** [comment]
I guess that it can be closed ?

**Pull request closed without merging**